pkg/controller/tektoninstallation/tektoninstallation_controller_test.go

```diff
@@ -60,10 +65,102 @@ func TestTektonInstallationController(t *testing.T) {
 				Exists().
 				HasSpec(tektonSub.Spec)
 
+			AssertThatTektonInstallation(t, tektonInstallation.Namespace, tektonInstallation.Name, cl).
+				HasConditions(Unknown())
+		})
+
+	})
+
+	// reconciling on tektonconfig resource watcher
+	t.Run("tektonconfig watcher", func(t *testing.T) {
+
+		t.Run("installed tekton installation", func(t *testing.T) {
+			// given
+			tektonInstallation := NewInstallation()
+			tektonConfig := NewTektonConfig("applied-addons", "installed", "validated-pipeline")
+			cl, r := configureClient(t, tektonInstallation,
+				NewSubscription(SubscriptionNamespace),
+				tektonConfig)
+			r.watchTektonConfig = func() error {
+				return nil
+			}
+			request := newReconcileRequest(tektonInstallation)
+
+			// when
+			_, err := r.Reconcile(request)
+
+			// then
+			require.NoError(t, err)
+			AssertThatSubscription(t, SubscriptionNamespace, SubscriptionName, cl).Exists()
```

Comment:
I guess that you don't have to verify the existence of the subscription resource - it's been created as part of the test-setup - applicable also for other cases